pymc3/distributions/timeseries.py

```diff
@@ -346,19 +305,18 @@ class MvGaussianRandomWalk(distribution.Continuous, _CovSet):
     def __init__(self, mu=0., cov=None, tau=None, chol=None, lower=True, init=Flat.dist(),
                  *args, **kwargs):
         super(MvGaussianRandomWalk, self).__init__(*args, **kwargs)
-        super(MvGaussianRandomWalk, self).__initCov__(cov, tau, chol, lower)
 
         self.mu = mu = tt.as_tensor_variable(mu)
         self.init = init
         self.mean = tt.as_tensor_variable(0.)
+        self.innovArgs = (self.mu, cov, tau, chol, lower)
+        self.innov = multivariate.MvNormal.dist(*self.innovArgs)
 
     def logp(self, x):
```

Comment:
This only works if x has rank 1, right? Maybe we should intercept the shape in `__init__` (one of the kwargs) and throw an error if that doesn't have length 1, just to give people a better error message.

Reply:
I checked to be sure: this only accepts x and cov/chol with ndim 2, and correctly controls that the shapes are compatible.

Reply:
Related observations:
- To be more precise, it controls/catches any glitch between shape and cov upon sampling, not at declaration time. That is also the behaviour of MvN/MvT. While that could be improved, I think it should then happen in those classes.
- Another question entirely (for later, perhaps) is wether we want to intercept any such errors here, as to make them more contextual for the timeseries classes.

However this is all probably not what your concern about the rank was?

Reply:
I should have said rank 2, not 1. I didn't actually think about the fact that mvnormal doesn't support higher rank input either, but I still think we should check this in this class. If at some point in the future we extend mvnormal, people wouldn't get an error then, but I'm not sure the results are what we want. (eg which dimension is the one with the correlations? In mvnormal it is the last one....)
Here is the code in mvnormal that I have in mind:
https://github.com/pymc-devs/pymc3/blob/master/pymc3/distributions/multivariate.py#L37

Reply:
Right, gotcha. If my understanding is correct, in the case of a random walk, `ndim=1` doesn't make sense either, right? So shape can only be two-dimensional for these distributions. 

So, as it stands, I would simply add this control in `__init__`:

        if len(self.shape) != 2:
            raise ValueError("Only 2 dimensions are allowed.")

Perhaps for a separate PR — when given 3 dims, we could (I think?) diff the obs separately along the time axis, concatenate/stack the thus obtained innovations to two dims, and pass that on to Mv.